### PR TITLE
nested nextflow schema params

### PR DESCRIPTION
**Pull request by mattheww95**
Created a nested nextflow_schema.json

### CODE REVIEW

File: nextflow_schema.json [hunk 1]
```diff
@@ -1,3015 +1,2233 @@
...
+      "$ref": "#/definitions/spades"
+    },
+    {
+      "$ref": "#/definitions/checkm"
+    },
+    {
+      "$ref": "#/definitions/kraken2"
+    }
+  ],
+  "chopper": {
```
Comment (apetkau) [inline nextflow_schema.json:1596]: Parameters that aren't under a section of `definitions` need to be under a `properties` object. That is everything under here needs to be written like:

```json
  "properties": {
      "chopper": {
      // ...
      },
  // ...
}
```

File: nextflow_schema.json [hunk 2]
```diff
@@ -1,3015 +1,2233 @@
...
+    "quality": {
```
Comment (apetkau) [inline nextflow_schema.json:1597]: Also, each nested parameter containing any sub parameters needs to be declared `"type": "object"` and have nested parameters under it's own "properties" section.

For example:

```json
"chopper": {
    "type": "object",
    "properties": {
        "quality": {
            "type": "integer",
            "default": 0,
            "hidden": true
        },
        "minlength": {
            "type": "integer",
            "default": 100,
            "hidden": true
       },
       // ...
    }
}
```

**Review by apetkau — CHANGES_REQUESTED** [review]
This looks amazing @mattheww95 . Thanks so much for all your hard work. I do have one comment for you.

File: nextflow_schema.json (hunk 1 above)
Comment (apetkau) [reply nextflow_schema.json:1596]: Also, each of these top-level nested parameter groups (like `chopper`) should be marked `"hidden": true`.

File: nextflow_schema.json (hunk 2 above)
Comment (mattheww95) [reply nextflow_schema.json]: This should be resolved in the newest upload

File: nextflow_schema.json (hunk 1 above)
Comment (mattheww95) [reply nextflow_schema.json]: fixed in latest commit! note, the hidden message appears underneath the nested block.

Comment (apetkau) [reply nextflow_schema.json]: Thank you. Due to the change to not include nested parameters in the schema, this is also no longer needed.

File: nextflow_schema.json (hunk 2 above)
Comment (apetkau) [reply nextflow_schema.json]: Thank you. Due to the change to not include nested parameters in the schema, this is also no longer needed.

**Review by apetkau — APPROVED** [review]
I have fixed up the tests and I have made some additional changes. The main issue is that Nextflow JSON Schemas don't support nested parameters, so we have to remove these from the JSON schema files (sorry about getting mixed up about that and thinking nested parameters would just be ignored).

So, I have made the following changes:
* Removed `nextflow_schema.json` and re-generated it from the config file using `nf-core schema build`
    * Re-structured schema to group common parameters together and added descriptions and appropriate minimums for integer parameters.
* Added test cases to verify that validation is happening for parameters and passing incorrect values fails the pipeline with an appropriate error message.
* Added tests to make sure that the identified species is set appropriately

I also tested how this would look in IRIDA Next (see below):

image

**Pull request merged**